### PR TITLE
pwninit: init at 3.2.0

### PR DESCRIPTION
**Pull request by spencerpogo**
###### Description of changes

Add https://github.com/io12/pwninit

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/pwninit/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ fetchFromGitHub, lib, rustPlatform, openssl, pkg-config, xz }:
+
+let version = "3.2.0";
+in rustPlatform.buildRustPackage {
+  pname = "pwninit";
+  inherit version;
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/pwninit/default.nix:6]:
```suggestion
rustPlatform.buildRustPackage rec {
  pname = "pwninit";
  version = "3.2.0";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by viraptor — COMMENTED** [review]
Result of `nixpkgs-review pr 176458` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>pwninit</li>
  </ul>
</details>

```
error: builder for '/nix/store/k2k0wz79d24nsafzwzbfp1xys7vinlca-pwninit-3.2.0.drv' failed with exit code 101;
       last 10 log lines:
       >    Compiling reqwest v0.11.9
       >    Compiling pwninit v3.2.0 (/private/tmp/nix-build-pwninit-3.2.0.drv-0/source)
       > error: linking with `/nix/store/rr7nsvr2x8ra1zd42rhwclpk71r0a63m-clang-wrapper-11.1.0/bin/cc` failed: exit status: 1
       >   |
       >   = note: "/nix/store/rr7nsvr2x8ra1zd42rhwclpk71r0a63m-clang-wrapper-11.1.0/bin/cc" "-m64" "-arch" "x86_64" "/private/tmp/nix-build-pwninit-3.2.0.drv-0/source/target/x86_64-apple-darwin/release/deps/pwninit-0c4e1ed6eb77b4ad.pwninit.b4bf9ee8-cgu.4.rcgu.o" "-L" "/private/tmp/nix-build-pwninit-3.2.0.drv-0/source/target/x86_64-apple-darwin/release/deps" "-L" "/private/tmp/nix-build-pwninit-3.2.0.drv-0/source/target/release/deps" "-L" "/nix/store/wdkckampbl0iz7kdl8w70b7cf63n0kcf-xz-5.2.5/lib" "-L" "/private/tmp/nix-build-pwninit-3.2.0.drv-0/source/target/x86_64-apple-darwin/release/build/zstd-sys-704978891f088fcc/out" "-L" "/nix/store/r7c3xwwpq0gyja1zsp37fpjfz15xn75n-rustc-1.61.0/lib/rustlib/x86_64-apple-darwin/lib" "/private/tmp/nix-build-pwninit-3.2.0.drv-0/rustcwmj6w4/libzstd_sys-afaecc0d3251bd5f.rlib" "/nix/store/r7c3xwwpq0gyja1zsp37fpjfz15xn75n-rustc-1.61.0/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-f13ca9014bed4545.rlib" "-framework" "Security" "-framework" "CoreFoundation" "-llzma" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/nix/store/r7c3xwwpq0gyja1zsp37fpjfz15xn75n-rustc-1.61.0/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/private/tmp/nix-build-pwninit-3.2.0.drv-0/source/target/x86_64-apple-darwin/release/deps/pwninit-0c4e1ed6eb77b4ad" "-Wl,-dead_strip" "-nodefaultlibs"
       >   = note: ld: framework not found Security
       >           clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       >
       >
       > error: could not compile `pwninit` due to previous error
       For full logs, run 'nix log /nix/store/k2k0wz79d24nsafzwzbfp1xys7vinlca-pwninit-3.2.0.drv'.
error: 1 dependencies of derivation '/nix/store/i902af63i45c9ivbl5v7sziiwl7i1mr6-review-shell.drv' failed to build
```

File: pkgs/development/tools/misc/pwninit/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "pwninit";
+    rev = version;
+    sha256 = "sha256-XKDYJH2SG3TkwL+FN6rXDap8la07icR0GPFiYcnOHeI=";
+  };
+
+  buildInputs = [
+    (lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security)
+    openssl
+    xz
+  ];
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/pwninit/default.nix:18]:
```suggestion
  buildInputs = [
    openssl
    xz
  ]
  ++ lib.optionals stdenv.isDarwin Security;
```

Again, do not bring darwin into the scope of the expression. List the dependencies directly.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -9600,6 +9600,8 @@ with pkgs;
 
   pwndbg = callPackage ../development/tools/misc/pwndbg { };
 
+  pwninit = callPackage ../development/tools/misc/pwninit { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:9603]:
```suggestion
  pwninit = callPackage ../development/tools/misc/pwninit {
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

File: pkgs/development/tools/misc/pwninit/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ darwin, fetchFromGitHub, lib, rustPlatform, stdenv, openssl, pkg-config, xz }:
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/pwninit/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, rustPlatform
, openssl
, pkg-config
, xz
, Security
}:
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 176458` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pwninit</li>
  </ul>
</details>

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**